utils/s3/client.cc

```diff
@@ -284,27 +266,28 @@ future<> client::do_retryable_request(group_client& gc, http::request req, http:
     }
 }
 
-http::experimental::client::reply_handler client::make_s3_reply_handler(
-        http::experimental::client::reply_handler&& handler, http::reply::status_type expected, multipart_upload_completion is_mpu_completion) {
-    return [is_mpu_completion, handler{std::move(handler)}, expected](const http::reply& rep, input_stream<char>&& in) -> future<> {
-        auto payload = std::move(in);
-        auto error = co_await look_for_error(rep, payload, is_mpu_completion);
-        if (error.get_error_type() != aws::aws_error_type::OK) {
-            co_await coroutine::return_exception(aws::aws_exception(std::move(error)));
-        }
-        if (rep._status != expected) {
-            co_return co_await coroutine::return_exception(httpd::unexpected_status_error(rep._status));
-        }
-        co_await handler(rep, std::move(payload));
-    };
-}
-
 future<> client::make_request(http::request req, http::experimental::client::reply_handler handle, http::reply::status_type expected) {
-    return make_request(std::move(req), [](http::experimental::client::reply_handler&& handler, http::reply::status_type expected) {
-            return make_s3_reply_handler(std::move(handler), expected, multipart_upload_completion::no);
-        },
-        std::move(handle),
-        expected);
+    return make_request(std::move(req),[](http::experimental::client::reply_handler&& handler, http::reply::status_type expected) {
+        return [handler{std::move(handler)}, expected](const http::reply& rep, input_stream<char>&& in) -> future<> {
```

Comment:
nit: we get more used to `handler = std::move(handler)` capturing syntax

Reply:
done